Sources/SpeziCLAID/CLAIDRuntime.swift

```diff
@@ -0,0 +1,170 @@
+    
+    
+    private let patchedCLAIDConfig: String = "patched_config.json"
+    private var patchedCLAIDConfigPath: String = ""
+    
+    //private let moduleRegistrationFunc: () async throws -> Void
+    // Reference to the PyCLAIDLoader module instance defined in "PyCLAIDLoader.py".
+    private var pyCLAIDLoader: PythonObject?
+    
+    private var config = Claidservice_CLAIDConfig()
+    private let hostName: String
+    private let userName: String
+    private let deviceId: String
+    
+    /// Retrieves the path to the CLAID configuration stub file inside the app bundle
+    /// - Returns: Path to the configuration file if found, otherwise throws an error
+    func getCLAIDConfigStubPath() throws -> String? {
+        guard let fileURL = Bundle.main.url(forResource: "claid_main_config", withExtension: "json") else {
+            throw CLAIDError("Could not find claid_main_config.json in bundle.")
+        }
+        return fileURL.path
+    }
+    
+    /// Loads the CLAID config from the specified path, then looks up the host and adds the additional Modules to it.
+    /// Finally, stores the new config to a different file
+    func writeOutCLAIDConfig(
+        config: Claidservice_CLAIDConfig
+    ) throws {
+        let tempDirectory = FileManager.default.temporaryDirectory
+        let tempOutputPath = tempDirectory.appendingPathComponent(patchedCLAIDConfig).path
+        self.patchedCLAIDConfigPath = tempOutputPath
+        
+        // Serialize back to JSON
+        let jsonData = try config.jsonUTF8Data()
+        
+        // Save to output path
+        try jsonData.write(to: URL(fileURLWithPath: self.patchedCLAIDConfigPath))
+    }
+    
+    /// Asynchronous initialization of CLAID using configuration stub
+    @MainActor
+    func startCLAIDFramework() async {
+        do {
+            print("Calling module registration func")
+            
+            for (moduleId, module) in CLAIDRuntime.moduleRegistry.preloadedCLAIDModules {
+                print("preloading \(moduleId) of type \(String(describing: type(of: module)))")
+                try await CLAID.registerModule(type(of: module))
+                await CLAID.addPreloadedModule(moduleId: moduleId, module: module)
+            }
+            
+           // try await self.moduleRegistrationFunc()
+            // Start CLAID with test configuration
+            try await CLAID.start(
+                configFile: patchedCLAIDConfigPath,
+                hostID: self.hostName,
+                userID: "test_user",
+                deviceID: "test_device"
+            )
+            
+            // Ensure pyCLAID is initialized and start it
+            guard let pyCLAIDLoader = self.pyCLAIDLoader else {
+                throw CLAIDError("Failed to start pyCLAID, object is null.")
+            }
+            _ = pyCLAIDLoader.attach_python_runtime()
+        } catch {
+            print("Error starting CLAID: \(error)")
+        }
```

Comment:
I think we should communicate this error back to the calling function to ensure that error are handled properly beyond print statements?